File: app/src/debug/java/theo/base/dagger/BuildVariantModule.java
package theo.base.dagger;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import theo.base.AppInitializer;
import theo.base.DebugAppInitializer;

@Module
public class BuildVariantModule {

    @Singleton
    @Provides
    AppInitializer provideDebugAppInitializer(){
        return new DebugAppInitializer();
    }
}

File: app/src/debug/kotlin/theo/base/dagger/BuildVariantModule.kt
package theo.base.dagger

import dagger.Module
import dagger.Provides
import theo.base.AppInitializer
import theo.base.DebugAppInitializer
import javax.inject.Singleton

@Module
class BuildVariantModule {

    @Singleton
    @Provides
    internal fun provideDebugAppInitializer(): AppInitializer {
        return DebugAppInitializer()
    }
}
